pkg/cli/sanitize/hide_secrets.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanitize
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"helm.sh/helm/v3/pkg/release"
+
+	"gopkg.in/yaml.v3"
```

Comment:
If you are going to use a YAML parser, please stick to the existing `sigs.k8s.io/yaml` parser used across Helm's codebase.

`sigs.k8s.io/yaml` is used across the Kubernetes ecosystem, and for better or worse has known compatibility issues with the v3 parser.

https://github.com/helm/helm/blob/e9e6b07f62e2bb5067afe79782ba3dd85efbd00e/go.mod#L47